core/src/WsTrust/WsTrustResponse.cs

```diff
@@ -0,0 +1,168 @@
+﻿//----------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml;
+using System.Xml.Linq;
+using Microsoft.Identity.Core;
+
+namespace Microsoft.Identity.Core.WsTrust
+{
+    internal class WsTrustResponse
+    {
+        public const string Saml1Assertion = "urn:oasis:names:tc:SAML:1.0:assertion";
+
+        public string Token { get; private set; }
+
+        public string TokenType { get; private set; }
+
+        public static WsTrustResponse CreateFromResponse(string response, WsTrustVersion version)
+        {
+            XDocument responseDocument = XDocument.Parse(response, LoadOptions.None); // Could throw XmlException
```

Comment:
The original code thows an AdalException in case the parsing fails - why not throw as well?

Reply:
OK now the exception mechanism in Core has been chosen, I'm making change at the caller of this method to address this comment. Consider it done! :-)

---
In reply to: [203679227](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1118#discussion_r203679227) [](ancestors = 203679227)